drivers/display/display_st7789v.c

```diff
@@ -161,7 +160,7 @@ static int st7789v_write(const struct device *dev,
 			 const struct display_buffer_descriptor *desc,
 			 const void *buf)
 {
-	struct st7789v_data *data = (struct st7789v_data *)dev->data;
+	struct st7789v_config *config = (struct st7789v_config *)dev->config;
```

Comment:
```suggestion
	struct st7789v_config *config = dev->config;
```

Reply:
@gmarull After applying the suggestion, the compiler complains about discarding 'const' qualifier. When making all references to `config` const I end up with a last warning for the spi_buf:
```
/home/casper/src/zephyrproject/zephyr/drivers/display/display_st7789v.c: In function 'st7789v_transmit':
/home/casper/src/zephyrproject/zephyr/drivers/display/display_st7789v.c:85:47: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
   85 |         const struct spi_buf tx_buf = {.buf = &cmd, .len = 1};
      |                                               ^
/home/casper/src/zephyrproject/zephyr/drivers/display/display_st7789v.c:93:47: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
   93 |         const struct spi_buf tx_buf = {.buf = tx_data, .len = tx_count};
      |                                               ^~~~~~~
```
Any ideas how to fix that?

Reply:
`struct st7789v_config *config` should actually be `const struct st7789v_config *config`

Reply:
I added the `const` modifier to the config struct. However I needed to add a `(void *)` to the tx_buf. I am unsure whether that is the best solution.